### PR TITLE
SSM update

### PR DESCRIPTION
**Pull request by thisismana**
- rewrote config management to use the _AWS Simple Systems Manager_.
This allows using a static config which will later be used to get rid of
the Play 2.7 global application
- consolidated capi modules: merged each `-client` module with their
corresponding `-core` module.
- Removed the `Env` for all the S3 services since noone used them.

### CODE REVIEW

File: core/src/main/scala/de/welt/contentapi/core/client/services/configuration/ApiConfiguration.scala [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  log.info(s"Environment loaded as: app=$app, stage=$stage")
+}
+
+/**
+  * provide configuration required by the WCAPI itself
+  *
+  * - Services accessing AWS/S3
+  * - API
+  */
+class ApiConfiguration extends Loggable {
```
Comment (thisismana) [inline core/src/main/scala/de/welt/contentapi/core/client/services/configuration/ApiConfiguration.scala]: Naming?

File: core/src/main/scala/de/welt/contentapi/core/client/services/configuration/Mode.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package de.welt.contentapi.core.client.services.configuration
+
+sealed trait Mode {
```
Comment (thisismana) [inline core/src/main/scala/de/welt/contentapi/core/client/services/configuration/Mode.scala]: Naming? `Mode`, `Stage`, ... ?

File: core/src/main/scala/de/welt/contentapi/core/client/services/aws/ssm/ParameterStore.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      .withRegion(region.getName)
+      .build()
+  } match {
+    case Success(value) ⇒ value
+    case Failure(th) ⇒
+      log.error("Could not initialize AWS SSM Client.", th)
+      throw th
+  }
+
+  def get(key: String): String = {
```
Comment (harryurban) [inline core/src/main/scala/de/welt/contentapi/core/client/services/aws/ssm/ParameterStore.scala]: Option or Try?

File: core/src/main/scala/de/welt/contentapi/core/client/services/configuration/ApiConfiguration.scala [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    for {
+      config ← Try(rootConfig.getConfig("build_info"))
+      key ← Try(config.getString("module"))
+      value ← Try(config.getList("dependencies").asScala.map(_.unwrapped())
+        .collect { case s: String ⇒ s }.sorted)
+        .orElse(Success(mutable.Buffer.empty[String]))
+    } yield key → value
+  }
+
+  protected[configuration] val findCurrentModule: Map[String, mutable.Buffer[String]] ⇒ Option[String] = projectSetup ⇒ projectSetup.find {
```
Comment (harryurban) [inline core/src/main/scala/de/welt/contentapi/core/client/services/configuration/ApiConfiguration.scala]: `projectSetup` -> irgendwas mit `sbt modules`

File: core/src/main/scala/de/welt/contentapi/core/client/services/configuration/ApiConfiguration.scala [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+      log.debug(s"Loading test configuration from $testConfigResource")
+      val config = configFromResource(testConfigResource)
+      log.debug(ApiConfiguration.configuration.root().render(ConfigRenderOptions.concise()))
+      config
+    } else {
+
+      val sw = Stopwatch.createStarted()
+
+      log.debug("Started loading Configuration.")
+      val userPrivate = configFromFile(s"${System.getProperty("user.home")}/.welt/frontend.conf")
```
Comment (harryurban) [inline core/src/main/scala/de/welt/contentapi/core/client/services/configuration/ApiConfiguration.scala]: document in readme

**Pull request merged**